Source/DafnyCore/Compilers/Python/Compiler-python.cs

```diff
@@ -177,20 +177,29 @@ class PythonCompiler : SinglePassCompiler {
       var relevantTypeParameters = typeParameters.Where(NeedsTypeDescriptor);
       var args = relevantTypeParameters.Comma(tp => tp.GetCompileName(Options));
       if (!string.IsNullOrEmpty(args)) { args = $", {args}"; }
+      var isNewtypeWithTraits = cls is NewtypeDecl { ParentTraits: { Count: > 0 } };
```

Comment:
No, with `/generalTraits:1`, it can be `true`. Here's an example input:

``` dafny
newtype MyInt extends Trait = int
trait Trait { }

method Main() {
  var m: MyInt := 12;
}
```

But without the new type system (engaged with `/typeSystemRefresh:1`, but not fully supported in this PR), you can't really make use of the parent traits of the newtype.

Reply:
Thanks!

Reply:
I'm fairly certain that this worked for me at the time, but doesn't seem to in this PR's current state.